### PR TITLE
resource_control: add processed_keys and ru to runaway

### PR DESCRIPTION
**Pull request by HuSharp**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/55147

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: information-schema/information-schema-runaway-watches.md [hunk 1]
```diff
@@ -117,3 +124,4 @@ RESOURCE_GROUP_NAME: rg1
     - `Exact` 表示按照 SQL 文本匹配，此时列 `WATCH_TEXT` 显示 SQL 文本。
 * `SOURCE`：识别项来源，如果是被 `QUERY_LIMIT` 规则识别，则显示识别到的 TiDB IP；如果是手动添加，则显示 `manual`。
 * `ACTION`：识别后的对应操作。
+* `RULE`：识别规则。目前包括 `ElapsedTime`, `ProcessedKeys` 和 `RequestUnit` 三种规则。格式 `ProcessedKeys = 666(10)`，其中 `666` 为实际值，`10` 为阈值。
```
Comment (lilin90) [inline information-schema/information-schema-runaway-watches.md:127]:
```suggestion
* `RULE`：识别规则。目前包括 `ElapsedTime`、`ProcessedKeys` 和 `RequestUnit` 这三种规则。格式为 `ProcessedKeys = 666(10)`，其中 `666` 为实际值，`10` 为阈值。
```

File: tidb-resource-control.md [hunk 2]
```diff
@@ -217,9 +217,11 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询（仅指 `S
 
 #### `QUERY_LIMIT` 参数说明
 
-支持的条件设置：
+超过以下任一条件限制时，查询会被识别为 Runaway Query：
 
-- `EXEC_ELAPSED`: 当查询执行的时间超限时，识别为 Runaway Query。
+- `EXEC_ELAPSED`: 检测查询执行的时间是否超限
+- `PROCESSED_KEYS`: 检测 Coprocessor 处理的 key 的数量是否超限
+- `REQUEST_UNIT`: 检测执行语句消耗的总读写 RU 是否超限
```
Comment (lilin90) [inline tidb-resource-control.md:224]:
```suggestion
- `EXEC_ELAPSED`：检测查询执行的时间是否超限
- `PROCESSED_KEYS`：检测 Coprocessor 处理的 key 的数量是否超限
- `REQUEST_UNIT`：检测执行语句消耗的总读写 RU 是否超限
```

File: tidb-resource-control.md [hunk 3]
```diff
@@ -217,9 +217,11 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询（仅指 `S
 
 #### `QUERY_LIMIT` 参数说明
 
-支持的条件设置：
+超过以下任一条件限制时，查询会被识别为 Runaway Query：
```
Comment (lilin90) [inline tidb-resource-control.md:220]:
```suggestion
如果查询超过以下任一限制，就会被识别为 Runaway Query：
```

File: tidb-resource-control.md [hunk 4]
```diff
@@ -244,6 +246,8 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询（仅指 `S
 | 参数            | 含义           | 备注                                   |
 |---------------|--------------|--------------------------------------|
 | `EXEC_ELAPSED`  | 当查询执行时间超过该值后被识别为 Runaway Query | EXEC_ELAPSED =`60s` 表示查询的执行时间超过 60 秒则被认为是 Runaway Query。 |
+| `PROCESSED_KEYS` | 当 Coprocessor 处理的 key 的数量超过该值后被识别为 Runaway Query | PROCESSED_KEYS = `1000` 表示 Coprocessor 处理的 key 的数量超过 1000 则被认为是 Runaway Query。 |
+| `REQUEST_UNIT`  | 当查询消耗的总读写 RU 超过该值后被识别为 Runaway Query | REQUEST_UNIT = `1000` 表示查询消耗的总读写 RU 超过 1000 则被认为是 Runaway Query。 |
```
Comment (lilin90) [inline tidb-resource-control.md:250]:
```suggestion
| `EXEC_ELAPSED`  | 当查询执行时间超过该值时，会被识别为 Runaway Query | `EXEC_ELAPSED = 60s` 表示查询的执行时间超过 60 秒则被认为是 Runaway Query。 |
| `PROCESSED_KEYS` | 当 Coprocessor 处理的 key 的数量超过该值时，查询会被识别为 Runaway Query | `PROCESSED_KEYS = 1000` 表示 Coprocessor 处理的 key 的数量超过 1000 则被认为是 Runaway Query。 |
| `REQUEST_UNIT`  | 当查询消耗的总读写 RU 超过该值时，查询会被识别为 Runaway Query | `REQUEST_UNIT = 1000` 表示查询消耗的总读写 RU 超过 1000 则被认为是 Runaway Query。 |
```

File: information-schema/information-schema-runaway-watches.md [hunk 5]
```diff
@@ -39,25 +40,28 @@ SELECT * FROM INFORMATION_SCHEMA.RUNAWAY_WATCHES\G
...
-RESOURCE_GROUP_NAME: rg2
-         START_TIME: 2023-07-28 01:45:30
-           END_TIME: UNLIMITED
-              WATCH: Similar
-         WATCH_TEXT: 3d48fca401d8cbb31a9f29adc9c0f9d4be967ca80a34f59c15f73af94e000c84
+                 ID: 2
+RESOURCE_GROUP_NAME: default
+         START_TIME: 2024-09-09 03:36:22
+           END_TIME: 2024-09-09 03:46:22
+              WATCH: Exact
```
Comment (glorv) [inline information-schema/information-schema-runaway-watches.md:59]: Better use 2 different type2 for these 2 example record.

Comment (HuSharp) [reply information-schema/information-schema-runaway-watches.md:59]: added

**Review by glorv — COMMENTED** [review]
LGTM

File: information-schema/information-schema-runaway-watches.md [hunk 6]
```diff
@@ -24,6 +24,7 @@ DESC RUNAWAY_WATCHES;
 | WATCH_TEXT          | text         | NO   |      | NULL    |       |
 | SOURCE              | varchar(128) | NO   |      | NULL    |       |
 | ACTION              | varchar(12)  | NO   |      | NULL    |       |
+| RULE                | varchar(128)  | NO   |      | NULL    |       |
 +---------------------+--------------+------+------+---------+-------+
 8 rows in set (0.00 sec)
```
Comment (lilin90) [inline information-schema/information-schema-runaway-watches.md:29]:
```suggestion
9 rows in set (0.00 sec)
```

File: information-schema/information-schema-runaway-watches.md [hunk 7]
```diff
@@ -24,6 +24,7 @@ DESC RUNAWAY_WATCHES;
 | WATCH_TEXT          | text         | NO   |      | NULL    |       |
 | SOURCE              | varchar(128) | NO   |      | NULL    |       |
 | ACTION              | varchar(12)  | NO   |      | NULL    |       |
+| RULE                | varchar(128)  | NO   |      | NULL    |       |
```
Comment (lilin90) [inline information-schema/information-schema-runaway-watches.md:27]:
```suggestion
| RULE                | varchar(128) | NO   |      | NULL    |       |
```

File: information-schema/information-schema-runaway-watches.md [hunk 8]
```diff
@@ -69,39 +83,52 @@ QUERY WATCH ADD RESOURCE GROUP rg1 SQL TEXT EXACT TO 'select * from sbtest.sbtes
 再次查询 Runaway Queries 识别名单：
 
 ```sql
-SELECT * FROM INFORMATION_SCHEMA.RUNAWAY_WATCHES\G
+SELECT * FROM INFORMATION_SCHEMA.RUNAWAY_WATCHES\G;
```
Comment (lilin90) [inline information-schema/information-schema-runaway-watches.md:86]: 文档里一般不同时使用 `\G` 和 `;`，之前专门改过，见 https://github.com/pingcap/docs/pull/17298#issue-2255726893
```suggestion
SELECT * FROM INFORMATION_SCHEMA.RUNAWAY_WATCHES\G
```

File: information-schema/information-schema-runaway-watches.md [hunk 9]
```diff
@@ -39,25 +40,38 @@ SELECT * FROM INFORMATION_SCHEMA.RUNAWAY_WATCHES\G
 输出结果如下：
 
 ```sql
+mysql> SELECT * FROM INFORMATION_SCHEMA.RUNAWAY_WATCHES ORDER BY id\G
```
Comment (lilin90) [inline information-schema/information-schema-runaway-watches.md:43]: 这一行语句移到 line 37 更合适？这里只放输出结果

**Review by lilin90 — APPROVED** [review]
/lgtm

File: tidb-resource-control.md [hunk 10]
```diff
@@ -217,9 +217,11 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询（仅指 `S
 
 #### `QUERY_LIMIT` 参数说明
 
-支持的条件设置：
+如果查询超过以下任一限制，就会被识别为 Runaway Query：
 
-- `EXEC_ELAPSED`: 当查询执行的时间超限时，识别为 Runaway Query。
+- `EXEC_ELAPSED`：检测查询执行的时间是否超限
```
Comment (hfxsd) [inline tidb-resource-control.md:222]: 这个要改成 ElapsedTime 吗？

File: sql-statements/sql-statement-create-resource-group.md [hunk 11]
```diff
@@ -49,6 +49,8 @@ ResourceGroupRunawayOptionList ::=
 
 DirectResourceGroupRunawayOption ::=
     "EXEC_ELAPSED" EqOpt stringLit
```
Comment (hfxsd) [inline sql-statements/sql-statement-create-resource-group.md:51]: 这个要改成 ElapsedTime 吗？

File: sql-statements/sql-statement-alter-resource-group.md [hunk 12]
```diff
@@ -49,6 +49,8 @@ ResourceGroupRunawayOptionList ::=
 
 DirectResourceGroupRunawayOption ::=
     "EXEC_ELAPSED" EqOpt stringLit
```
Comment (hfxsd) [inline sql-statements/sql-statement-alter-resource-group.md:51]: 这个要改成 ElapsedTime 吗？

File: tidb-resource-control.md [hunk 13]
```diff
@@ -243,7 +245,9 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询（仅指 `S
 
 | 参数            | 含义           | 备注                                   |
 |---------------|--------------|--------------------------------------|
-| `EXEC_ELAPSED`  | 当查询执行时间超过该值后被识别为 Runaway Query | EXEC_ELAPSED =`60s` 表示查询的执行时间超过 60 秒则被认为是 Runaway Query。 |
+| `EXEC_ELAPSED`  | 当查询执行时间超过该值时，会被识别为 Runaway Query | `EXEC_ELAPSED = 60s` 表示查询的执行时间超过 60 秒则被认为是 Runaway Query。 |
```
Comment (hfxsd) [inline tidb-resource-control.md:248]: 这个要改成 ElapsedTime 吗？

File: tidb-resource-control.md (hunk 10 above)
Comment (HuSharp) [reply tidb-resource-control.md:222]: 考虑兼容性的话，这里不会改成 EXEC_ELAPSED

Comment (hfxsd) [reply tidb-resource-control.md:222]: ok

**Review by hfxsd — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Review by lilin90 — COMMENTED** [review]
/approve

File: tidb-resource-control.md [hunk 14]
```diff
@@ -244,6 +246,9 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询（仅指 `S
 
 | 参数            | 含义           | 备注                                   |
 |---------------|--------------|--------------------------------------|
+| `EXEC_ELAPSED`  | 当查询执行时间超过该值时，会被识别为 Runaway Query | `EXEC_ELAPSED = 60s` 表示查询的执行时间超过 60 秒则被认为是 Runaway Query。 |
+| `PROCESSED_KEYS` | 当 Coprocessor 处理的 key 的数量超过该值时，查询会被识别为 Runaway Query | `PROCESSED_KEYS = 1000` 表示 Coprocessor 处理的 key 的数量超过 1000 则被认为是 Runaway Query。 |
+| `RU`  | 当查询消耗的总读写 RU 超过该值时，查询会被识别为 Runaway Query | `RU = 1000` 表示查询消耗的总读写 RU 超过 1000 则被认为是 Runaway Query。 |
 | `EXEC_ELAPSED`  | 当查询执行时间超过该值后被识别为 Runaway Query | EXEC_ELAPSED =`60s` 表示查询的执行时间超过 60 秒则被认为是 Runaway Query。 |
```
Comment (lilin90) [inline tidb-resource-control.md:252]:
```suggestion
```

**Pull request merged**